### PR TITLE
feat: Remove mobile auth flow in favor of desktop auth flow

### PR DESCRIPTION
**Pull request by williardx**
https://artsyproduct.atlassian.net/browse/GRO-62
cc @artsy/grow-devs 

## Problem

Our mobile web auth flows break up the steps for each flow over multiple views, leading to a poor user experience and lower conversion. 

The original intent behind this was to follow best practices for forms on mobile web where steps are typically broken up to create a better user experience. However, due to a bug some parts of mobile web use a responsive version of the desktop auth flow which does not break up the form into multiple steps. The desktop auth flow on mobile web converts better than the original mobile web auth flow.

## Solution

Remove the mobile web auth flow and replace it with the desktop auth flow. This also has the benefit of simplifying our auth code and creating a more consistent user experience.

## What changed

The desktop auth flow will be the only auth flow on Artsy mobile web (except for inquiries, which is a separate case).
- Removes mobile web specific auth components in favor of using desktop auth components
- Removes most mobile specific helpers in `openAuthModal.ts` in favor of desktop versions where possible

Ideally we'd use the auth modal instead of explicit auth routes wherever we can, including in places like the Signup and Login buttons in the mobile nav menu. However, the mobile nav menu is itself already a modal and using another auth modal on top of it creates a poor user experience. I decided to keep the Signup and Login buttons as links to the `/signup` and /`login` routes.

## QA

- [x] User can successfully signup on `/signup`
- [x] User can successfully login on `/login`
- [x] User can successfully reset password on `/forgot`
- [x] Auth flows work on follow in mobile web
- [x] No regressions in auth modals in mobile web
- [x] No regressions in desktop auth flow

## Screenshots

<details>
  <summary>Screenshots </summary>

**Old Signup**

Screen Shot 2021-01-08 at 11 13 48 AM

**New Signup**

Screen Shot 2021-01-08 at 12 46 49 PM

**Old Login**

Screen Shot 2021-01-08 at 12 46 16 PM

**New Login**

Screen Shot 2021-01-08 at 12 44 09 PM

**Old Forgot**

Screen Shot 2021-01-08 at 11 20 45 AM

**New Forgot**

Screen Shot 2021-01-08 at 11 21 03 AM
</details>

### CODE REVIEW

**Comment by williardx** [comment]
Keeping in draft mode while I fix a few things and do QA.

**Comment by williardx** [comment]
Failing Cypress test is unrelated.

**Review by damassi — APPROVED** [review]
This all seems pretty straight forward 👍 

One question however: have you rebased against master? We removed the object sort keys plugin due to all the PR review churn (and inability to apply it globally due to obscure errors), wondering if you're out of sync.

**Comment by williardx** [comment]
@damassi I just rebased against master and pushed and now I don't see those changes being made anymore.

**Comment by damassi** [comment]
👌

File: src/v2/Components/Authentication/FormSwitcher.tsx [hunk 1]
```diff
@@ -181,13 +171,13 @@ export class FormSwitcher extends React.Component<FormSwitcherProps, State> {
...
-        Form = isMobile ? MobileLoginForm : LoginForm
+        Form = LoginForm
         break
       case ModalType.signup:
-        Form = isMobile ? MobileSignUpForm : SignUpForm
+        Form = SignUpForm
         break
       case ModalType.forgot:
-        Form = isMobile ? MobileForgotPasswordForm : ForgotPasswordForm
+        Form = ForgotPasswordForm
```
Comment (jonallured) [inline src/v2/Components/Authentication/FormSwitcher.tsx:180]: **suggestion [if-easy]:** extract map

I have a preference to see a construct like this done with a map, something like this:

```ts
const modalTypeMap = {
  ModalType.login: LoginForm,
  ModalType.signup: SignUpForm,
  ModalType.forgot: ForgotPasswordForm
}

// ...

const Form = modalTypeMap[this.state.type]
```

What do you think?

Note: unless I'm off here, I believe we have a `default` that returns `null` but then in the rendering we don't guard against it? Maybe it doesn't matter in reality, but seems like a case that's not being handled and I realize that covering this is not strictly speaking a part of this PR. Just wanted to point out something I noticed while reading this file!!

*format via [Conventional Comments](https://conventionalcomments.org/)*

**Review by jonallured — APPROVED** [review]
Just had one `if-easy` suggestion but this is awesome, nice one!!! 🤗

File: src/v2/Components/Authentication/FormSwitcher.tsx (hunk 1 above)
Comment (williardx) [reply src/v2/Components/Authentication/FormSwitcher.tsx:180]: Agreed that that's better! Is it all right if I leave that as a followup item for a separate auth code refactor? Other parts of the code would benefit from this.

**Pull request merged**